trussworks/react-uswds
# feat!: New Component: IconList

# Summary

Add new component: [IconList](https://designsystem.digital.gov/components/icon-list/)

## Related Issues or PRs

closes #1234
~~fixes a typo in one exported Icon in de855d8d4335645eef6a12a108ac7483ca7438df~~ Since the Icon API was changed to ensure no conflict between `IconList` and the `Icon`, `List`, (previous `IconList`, now `Icon.List`), this is a BREAKING CHANGE.
- When merging and squashing, add the following text:
  > BREAKING CHANGE: Icons no longer follow the pattern `Icon{ICON_NAME}`. They have are now prefixed components using "dot" notation as in `Icon.{ICON_NAME}`. The import therefore also changes, as you now only need to `import { Icon } ...`.
- Note that the side effects of this change are less efficient tree shaking. A consumer using any icon will effectively be importing ALL of the icons.

## How To Test

- Start up storybook (or use the ephemerally deployed storybook instance) and view/inspect the IconList examples.
- Verify Unit test coverage is acceptable and passing

### Screenshots (optional)

![image](https://user-images.githubusercontent.com/15805554/137199270-b20ae001-1208-4765-9ca7-d0d4db765504.png)

![image](https://user-images.githubusercontent.com/15805554/137199293-610f1660-a1d4-4533-ab96-5cbb1315799b.png)

![image](https://user-images.githubusercontent.com/15805554/137199325-2e793dc5-ff1f-436f-8d49-e7b9af4372cb.png)

![image](https://user-images.githubusercontent.com/15805554/137199340-96b95601-3d51-4947-b026-94114b57bd97.png)

![image](https://user-images.githubusercontent.com/15805554/137199354-afd0ce86-8aa1-4233-a497-65d4a04b1535.png)
